### ISSUE TITLE
Cannot build x86_64 module with nw-gyp

### ISSUE THREAD

**Issue by btjones** [nwjs/nw-gyp#39]
tl;dr;
I can't seem to change the architecture of a node module built with nw-gyp to x86_64, it always comes out as i386.

---

I've built node-webkit (0.10.1) as x86_64 (64 bit support recently added: https://github.com/rogerwang/node-webkit/issues/296):

```
$ lipo -info node-webkit.app/Contents/MacOS/node-webkit 

Non-fat file: node-webkit.app/Contents/MacOS/node-webkit is architecture: x86_64
```

I am trying to build a node module as x86_64 with nw-gyp. I've been testing with the hello-world module (https://github.com/joyent/node/tree/master/test/addons/hello-world) but I've tried others as well. No matter what I do, it comes out as i386.

```
$ cd node_modules/hello-world
$ nw-gyp rebuild --arch=x86_64 --target=0.10.1 --verbose

gyp info it worked if it ends with ok
gyp verb cli [ 'node',
gyp verb cli   '/usr/local/bin/nw-gyp',
gyp verb cli   'rebuild',
gyp verb cli   '--arch=x86_64',
gyp verb cli   '--target=0.10.1',
gyp verb cli   '--verbose' ]
gyp info using nw-gyp@0.12.2
gyp info using node@0.10.26 | darwin | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python /usr/bin/python
gyp verb check python version `python -c "import platform; print(platform.python_version());"` returned: "2.7.5\n"
gyp verb get node dir compiling against --target node version: 0.10.1
gyp verb command install [ '0.10.1' ]
gyp verb install input version string "0.10.1"
gyp verb installing legacy version? false
gyp verb install installing version: 0.10.1
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 0.10.1
gyp verb build dir attempting to create "build" dir: /Users/btj/projects/testapp/node_modules/hello-world/build
gyp verb build dir "build" dir needed to be created? /Users/btj/projects/testapp/node_modules/hello-world/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /Users/btj/projects/testapp/node_modules/hello-world/build/config.gypi
gyp verb config.gypi checking for gypi file: /Users/btj/projects/testapp/node_modules/hello-world/config.gypi
gyp verb common.gypi checking for gypi file: /Users/btj/projects/testapp/node_modules/hello-world/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn python
gyp info spawn args [ '/usr/local/lib/node_modules/nw-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/btj/projects/testapp/node_modules/hello-world/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/nw-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/btj/.nw-gyp/0.10.1/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/btj/.nw-gyp/0.10.1',
gyp info spawn args   '-Dmodule_root_dir=/Users/btj/projects/testapp/node_modules/hello-world',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp verb build type Release
gyp verb architecture x86_64
gyp verb node dev dir /Users/btj/.nw-gyp/0.10.1
gyp verb `which` succeeded for `make` /usr/bin/make
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  c++ '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/Users/btj/.nw-gyp/0.10.1/src -I/Users/btj/.nw-gyp/0.10.1/deps/uv/include -I/Users/btj/.nw-gyp/0.10.1/deps/v8/include  -Os -gdwarf-2 -mmacosx-version-min=10.5 -arch i386 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/binding/binding.o.d.raw  -c -o Release/obj.target/binding/binding.o ../binding.cc
  ./gyp-mac-tool flock ./Release/linker.lock c++ -bundle -Wl,-search_paths_first -mmacosx-version-min=10.5 -arch i386 -L./Release  -o Release/binding.node Release/obj.target/binding/binding.o -undefined dynamic_lookup
  SOLINK_MODULE(target) Release/binding.node: Finished
gyp info ok 
```

Build succeeds but the architecture is i386

```
$ lipo -info build/Release/binding.node 

Non-fat file: build/Release/binding.node is architecture: i386
```

If I try to run the i386 version I get this error (same error I get at runtime if I build my app with the i386 version):

```
$ node build/Release/binding.node 

module.js:356
  Module._extensions[extension](this, filename);
                               ^
Error: dlopen(/Users/btj/projects/testapp/node_modules/hello-world/build/Release/binding.node, 1): no suitable image found.  Did find:
    /Users/btj/projects/testapp/node_modules/hello-world/build/Release/binding.node: mach-o, but wrong architecture
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
```

I'm probably missing something simple... Any ideas? Thanks!
